src/satisfiability/planet.jl

```diff
@@ -132,31 +131,39 @@ function max_slack(x::Vector{<:Vector}, act)
     return (m, index)
 end
 
-# Only use tighten_bounds for feasibility check
-function tighten_bounds(problem::Problem, optimizer)
+function tighten_bounds(problem::Problem, optimizer;
+                        bounds::Vector{Hyperrectangle} = get_bounds(problem, false),
+                        encoding = TriangularRelaxedLP())
+    new_bounds = copy(bounds)
     model = Model(optimizer)
-    model[:bounds] = bounds = get_bounds(problem)
-    z = init_vars(model, problem.network, :z, with_input=true)
+    init_model_vars(model, problem, encoding, bounds=new_bounds, before_act=true)
```

Comment:
alternatively, we could have `bounds = copy(bounds)`, and then `new_bounds =model[:bounds]` after. Whichever is clearer.

Reply:
oh because we'll be updating them inside the model and we can just pull those out at the end?

Reply:
I tried updating this, I think it makes it cleaner